packages/ui/src/components/Form/bean/NewBeanModal.tsx

```diff
@@ -17,13 +18,19 @@ export type NewBeanModalProps = {
 export const NewBeanModal: FunctionComponent<NewBeanModalProps> = (props: NewBeanModalProps) => {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const [beanModel, setBeanModel] = useState<any>();
+  const submitRef = useRef<CustomAutoFormRef>(null);
 
   useEffect(() => {
     setBeanModel(props.beanName ? { name: props.beanName } : {});
   }, [props.beanName]);
 
-  const handleConfirm = useCallback(() => {
-    props.onCreateBean(beanModel as RegistryBeanDefinition);
+  const handleConfirm = useCallback(async () => {
+    // validation updates the bean model, so we need to clone it to avoid creating the bean with default values
+    const beanModelTmp = { ...beanModel };
```

Comment:
In this case, we're just copying the outer object, if `beanModel` has nested properties, those would still have the new value, so should use either `cloneDeep` from `lodash` or [structuredClone](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone).

Reply:
sure, sounds like this could cause issues - updated